tests/run-test.sh

```diff
@@ -19,7 +19,7 @@ set -euo pipefail
 
 function fail()
 {
-  sed -n '/^#$/,/^$/p' "${BASH_SOURCE[0]}"
+  sed -n "/^#$/,/^$/p" "${BASH_SOURCE[0]}"
```

Comment:
Please don't apply this part; if this still works as intended, then it's relying on subtle details of how `$` expands or doesn't expand as a variable reference inside double quotes. It will be easier to understand with single quotes.

Unlike Python, in shell script there is a functional difference between single and double quotes which matters a lot.

Reply:
Woops, I didn't notice the `$` signs... Fixed.